planetiler-core/src/main/java/com/onthegomap/planetiler/expression/Expression.java

```diff
@@ -359,11 +387,18 @@ public boolean evaluate(SourceFeature input, List<String> matchKeys) {
 
     @Override
     public String toString() {
-      return "matchAny(" + Format.quote(field) + ", " + values.stream().map(Format::quote)
+      return "matchAny(" + Format.quote(field) + ", " + values.stream().map(Expression::stringifyTypedObject)
```

Comment:
This doesn't encode the getter to use so we'll lose that when it's serialized. Only generate.java needs code to compile from `toString` so we could also rename this to `generateCode` and if the getter is not default then throw an unsupported operation exception?

Reply:
@msbarry would it be easier to just undo the toString() stuff since generate.java is only ever going to invoke the non-typed version of Expression?
\